hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/catalog/FederationDescriptionReader.java

```diff
@@ -208,27 +207,33 @@ else if ( addr.isURIResource() ) {
 	}
 
 	/**
-	 * Checks whether given RDF resource (fm) representing a federation
-	 * member is associated with a vocabulary mapping in the given federation
-	 * description (fd) and, if so, parses this vocabulary mapping and returns
-	 * it. Otherwise, this function returns <code>null</code>.
+	 * Attempts to retrieve and parse the vocabulary mapping associated with the
+	 * given RDF resource {@code fm}, representing a {@link FederationMember}, in
+	 * the given federation description {@code fd}.
+	 *
+	 * The method attempts to load the vocabulary mapping from the specified path or
+	 * URL and caches the result for reuse. If no vocabulary mappings file is
+	 * present, the method returns {@code null}.
+	 *
+	 * @param fm RDF resource for the federation member
+	 * @param fd RDF model of the federation description
+	 * @return parsed {@link VocabularyMapping}, or {@code null} if not specified
+	 * @throws org.apache.jena.riot.RiotNotFoundException if the mapping file cannot be loaded or parsed
 	 */
 	protected VocabularyMapping parseVocabMapping( final Resource fm, final Model fd ) {
-		if( fd.contains(fm, FD.vocabularyMappingsFile) ){
-			final RDFNode pathToMappingFile = fd.getRequiredProperty(fm, FD.vocabularyMappingsFile).getObject();
-
-			final String path = pathToMappingFile.toString();
-			if ( verifyValidVocabMappingFile(path) ) {
-				VocabularyMapping vm = vocabMappingByPath.get( path );
-				if ( vm == null ) {
-					vm = new VocabularyMappingWrappingImpl(path);
-					vocabMappingByPath.put( path, vm );
-				}
-				return vm;
-			}
+		if ( ! fd.contains( fm, FD.vocabularyMappingsFile ) ) {
+			return null;
 		}
 
-		return null;
+		final RDFNode pathToMappingFile = fd.getRequiredProperty( fm, FD.vocabularyMappingsFile ).getObject();
+
+		final String path = pathToMappingFile.toString();
+		VocabularyMapping vm = vocabMappingByPath.get( path );
+		if ( vm == null ) {
+			vm = new VocabularyMappingWrappingImpl( path );
```

Comment:
This constructor may throw an exception of type `RiotNotFoundException` (perhaps other ones as well?), but that is not clear from the code here. In other parts of `FederationDescriptionReader` we are throwing `IllegalArgumentException` right? In this sense, for the sake of consistency and transparency, I would prefer this line to be wrapped into a try-catch block where the catch part throws an `IllegalArgumentException` that wraps the exception that was caught.

An even better way would be to remove the constructor of `VocabularyMappingWrappingImpl` that is used here and, instead, do the `RDFDataMgr.loadGraph(path)` directly here, wrapped into a try-catch block as mentioned above.

Reply:
Okay, sounds good. Iäll update the code to load the graph here and use the `VocabularyMappingWrappingImpl` constructor that takes in a `Graph` argument instead, and wrap `RDFDataMgr.loadGraph(path)` with the try-except as suggested.